internal/dag/gatewayapi_processor.go

```diff
@@ -1432,6 +1432,13 @@ func (p *GatewayAPIProcessor) computeGRPCRoute(route *gatewayapi_v1alpha2.GRPCRo
 			})
 		}
 
+		// If no matches are specified, the implementation MUST match every gRPC request.
+		if rule.Matches == nil {
```

Comment:
Small nit, this will handle both a nil and an empty slice:
```suggestion
		if len(rule.Matches) == 0 {
```